### PR TITLE
Update bundled JDK to 11.0.15+10

### PR DESCRIPTION
**Pull request by reta**
Signed-off-by: Andriy Redko <andriy.redko@aiven.io>

### Description
Update bundled JDKs to addresses recently disclosed/found CVEs
 
### Issues Resolved
Part of https://github.com/opensearch-project/OpenSearch/issues/3026
 
### Check List
- [ ] New functionality includes testing.
  - [ ] All tests pass
- [ ] New functionality has been documented.
  - [ ] New functionality has javadoc added
- [X] Commits are signed per the DCO using --signoff 

By submitting this pull request, I confirm that my contribution is made under the terms of the Apache 2.0 license.
For more information on following Developer Certificate of Origin and signing off your commits, please check [here](https://github.com/opensearch-project/OpenSearch/blob/main/CONTRIBUTING.md#developer-certificate-of-origin).

### CODE REVIEW

**Comment by reta** [comment]
start grade check

**Comment by reta** [comment]
start gradle check

File: buildSrc/src/main/java/org/opensearch/gradle/test/DistroTestPlugin.java [hunk 1]
```diff
@@ -77,7 +77,7 @@
 public class DistroTestPlugin implements Plugin<Project> {
     private static final String SYSTEM_JDK_VERSION = "8u242+b08";
     private static final String SYSTEM_JDK_VENDOR = "adoptopenjdk";
-    private static final String GRADLE_JDK_VERSION = "17.0.2+8";
+    private static final String GRADLE_JDK_VERSION = "11.0.15+10";
```
Comment (dblock) [inline buildSrc/src/main/java/org/opensearch/gradle/test/DistroTestPlugin.java:80]: I presume this is unused since it didn't break anything?

Comment (reta) [reply buildSrc/src/main/java/org/opensearch/gradle/test/DistroTestPlugin.java:80]: To be honest, it seems like the version is used in some tests but not JDK itself, thanks!

**Comment by kartg** [comment]
Strange, the 1.3 backport workflow never kicked off for this.

**Comment by reta** [comment]
> Strange, the 1.3 backport workflow never kicked off for this. I don't think this fix has made it into 1.3.2

Wow, unexpected, will submit manually tomorrow first thing, thanks @kartg

**Comment by kartg** [comment]
@reta all good, I opened a manual backport and got it merged 👍  #3134

**Comment by reta** [comment]
> @reta all good, I opened a manual backport and got it merged 👍 #3134

Awesome, thanks a lot @kartg !

**Pull request merged**